docs/Style-Guide.md

```diff
@@ -0,0 +1,274 @@
+}
+```
+
+---
+
+## Components
+
+- Always use functional components, no classes.
+- Only **one** functional component per file unless the component is stateless, [eslint/no-multi-comp](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-multi-comp.md).
+- Stateless components should be in a seperate file is reused.
+- Component file names should be in PascalCase, `ExampleInputComponent.tsx`
+
+Components should follow this general structure.
+
+```tsx
+// imports
+import React from "react";
+
+import logo from "./logo.svg";
+import "./App.css";
+
+// constants
+const DEFAULT_VALUE = 5;
+
+// props / types
+interface ExampleProps {
+  onChange: (n: number) => void;
+}
+
+// component
+function Counter(): React.Element {
+  // hooks
+  const [count, setCount] = useState(DEFAULT_VALUE);
+
+  // logic
+  const onClickHandler = () => {
+    setCount(count + 1);
+  };
+
+  const isOdd = count % 2 !== 0;
+
+  return (
+    <div>
+      <button onClick={onClickHandler}>{t`click to increment`}</button>
+      <div>{count}</div>
+      {isOdd && <div>{t`value is odd!`}</div>}
+    </div>
+  );
+}
+
+// export
+export default Example;
+```
+
+### Takeaways
+
+- Use the `function` keyword declaration for components
+- Prop interface should be defined above the component
+
+### Good Examples
+
+- example
+- example
+- example
+- example
+- example
+
+Props should be spread in the function parameters
+
+```ts
+/* Button.tsx */
+
+interface ButtonProps {
+  text: string;
+  onClick: () => void;
+}
+
+// Good
+function Button({ text, onClick }: ButtonProps): React.Element {
+  return <button onClick={onClick}>{text}</button>;
+}
+
+// Bad
+function Button(props: ButtonProps): React.Element {
+  const { onClick, text } = props;
+  return <button onClick={onClick}>{text}</button>;
+}
+```
+
+Prop order should have nullables sorted last.
+
+```ts
+// Good
+interface ExampleProps {
+  position: string;
+  onClick: () => {};
+  title?: string;
+}
+
+// Bad
+interface ExampleProps {
+  position: string;
+  title?: string;
+  onClick: () => {};
+}
+```
+
+---
+
+## Function Syntax
+
+Standalone functions should be defined using the `function` keyword.
+
+```ts
+/* add.ts */
+
+// Good
+function add(a: number, b: number) {
+  return a + b;
+}
+
+// Bad
+const add = (a: number, b: number) => {
+  return a + b;
+};
+```
+
+Inline functions (callbacks, anonymous, props, etc.) should use `arrow function` syntax.
+
+```tsx
+/* Button.tsx */
+
+// Good
+function Button(): React.Element {
+  return (
+    <button
+      onClick={() => {
+        const { data } = fetchData();
+        setData(data);
+      }}
+    >
+      Click me
+    </button>
+  );
+}
+
+// Bad
+function Button(): React.Element {
+  return (
+    <button
+      onClick={function () {
+        const { data } = fetchData();
+        setData(data);
+      }}
+    >
+      Click me
+    </button>
+  );
+}
+```
+
+Anonymous functions should be reduced down to one line.
```

Comment:
I don't fully agree with this one as a rule-- Python lambda functions are enforced to be one-line and they get nasty fast.

I'm open to just not having a rule about this particular aspect of the codebase. In general, I prefer the curly-braces style unless it's something obviously one-linery, (like an simple anon function that calls a setState with some value in a click handler)